packages/core/src/components/floating-bar/floating-bar.tsx

```diff
@@ -0,0 +1,137 @@
+ * FloatingBar.PositionerPrimitive
+ * -----------------------------------------------------------------------------------------------*/
+
+/**
+ * Fixed position styles for the FloatingBar
+ */
+const positions = { top: 'initial', left: '50%', transform: 'translateX(-50%)' };
+
+export const FloatingBarPositionerPrimitive = forwardRef<
+    HTMLDivElement,
+    FloatingBarPositionerPrimitive.Props
+>((props, ref) => {
+    const { style, className, ...componentProps } = resolveStyles(props);
+
+    return (
+        <Popover.Positioner
+            ref={ref}
+            positionMethod="fixed"
+            style={{ ...style, ...positions }}
+            className={clsx(styles.positioner, className)}
+            {...componentProps}
+        />
+    );
+});
+FloatingBarPositionerPrimitive.displayName = 'FloatingBar.PositionerPrimitive';
+
+/* -------------------------------------------------------------------------------------------------
+ * FloatingBar.PopupPrimitive
+ * -----------------------------------------------------------------------------------------------*/
+
+export const FloatingBarPopupPrimitive = forwardRef<
+    HTMLDivElement,
+    FloatingBarPopupPrimitive.Props
+>((props, ref) => {
+    const { className, ...componentProps } = resolveStyles(props);
+
+    return (
+        <Popover.Popup ref={ref} className={clsx(styles.popup, className)} {...componentProps} />
+    );
+});
+FloatingBarPopupPrimitive.displayName = 'FloatingBar.PopupPrimitive';
+
+/* -------------------------------------------------------------------------------------------------
+ * FloatingBar.Popup
+ * -----------------------------------------------------------------------------------------------*/
+
+export const FloatingBarPopup = forwardRef<HTMLDivElement, FloatingBarPopup.Props>((props, ref) => {
+    return (
+        <FloatingBarPortalPrimitive>
+            <FloatingBarPositionerPrimitive>
+                <FloatingBarPopupPrimitive ref={ref} {...props} />
+            </FloatingBarPositionerPrimitive>
+        </FloatingBarPortalPrimitive>
+    );
+});
+FloatingBarPopup.displayName = 'FloatingBar.Popup';
+
+/* -----------------------------------------------------------------------------------------------*/
+
+export namespace FloatingBarRoot {
+    export type Props = Popover.Root.Props;
+}
+
+export namespace FloatingBarTrigger {
+    export type Props = VComponentProps<typeof Popover.Trigger>;
+}
+
+export namespace FloatingBarClose {
+    export type Props = VComponentProps<typeof Popover.Close>;
+}
+
+export namespace FloatingBarPortalPrimitive {
+    export type Props = VComponentProps<typeof Popover.Portal>;
+}
+
+export namespace FloatingBarPositionerPrimitive {
+    export type Props = VComponentProps<'div'>;
+}
+
+export namespace FloatingBarPopupPrimitive {
+    export type Props = VComponentProps<typeof Popover.Popup>;
+}
+
+export namespace FloatingBarPopup {
+    export type Props = FloatingBarPopupPrimitive.Props;
+}
```

Comment:
[Suggestion] I recommend replacing the namespace usage with standard named type exports.
 
- Standardization: Modern TypeScript development strongly prefers ES Modules over namespaces. As per the [TypeScript Handbook](https://www.typescriptlang.org/docs/handbook/namespaces-and-modules.html) and [eslint-plugin-typescript](https://typescript-eslint.io/rules/no-namespace/), namespaces are considered legacy for code organization.
-  Bundle Size: Namespaces compile to IIFEs or objects in JavaScript, which can hurt tree-shaking and add unnecessary runtime overhead compared to zero-cost type aliases or interfaces.

Proposed Change: Instead of `FloatingBarRoot.Props`, consider using a flat naming convention like `FloatingBarRootProps`.